src/firefly/js/ui/tap/SpatialSearch.jsx

```diff
@@ -92,13 +92,13 @@ export function SpatialSearch({cols, serviceUrl, serviceLabel, columnsModel, tab
 
     const updatePanelStatus= makePanelStatusUpdater(checkHeaderCtl.isPanelActive(), Spatial);
 
-    useEffect(() => {
+    /*useEffect(() => {
         if (canUpload) {
             if (uploadInfo) setVal(SPATIAL_TYPE,MULTI);
             return;
         }
         setVal(SPATIAL_TYPE,SINGLE);
-    }, [serviceUrl,canUpload]);
+    }, [serviceUrl,canUpload]);*/
```

Comment:
What is this commented out and what behavior does it change?

Reply:
Let's say you have IRSA selected as the TAP service. If you switch from `Single Shape` to `Multi-object` once, and then switch back to `Single Shape`. Now if you switch to, say, CADC, this useEffect gets triggered (because of `serviceUrl` changing). And since an uploadInfo object exists, `Multi-object` will be selected here by default instead of `Single Shape` (which was the user's last selection). 

Commenting out the useEffect gets rid of this behavior, while not really affecting anything else.  

Actually, it does cause an issue switching from a `Multi-object` selection to a service that doesn't support uploads like NED. I think this is the final useEffect we'll need here: 

```
useEffect(() => {
        if (!canUpload) setVal(SPATIAL_TYPE,SINGLE);
    }, [serviceUrl,canUpload]);
```

I'll push this change in my next commit so you can take a look.

Reply:
Thanks for helping me understand, the change makes sense.